libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -542,15 +543,22 @@ private void initializeLocationComponent(MapView mapView, MapboxMap map) {
     map.setMinZoomPreference(NAVIGATION_MINIMUM_MAP_ZOOM);
     map.setMaxZoomPreference(NAVIGATION_MAXIMUM_MAP_ZOOM);
     Context context = mapView.getContext();
-    int locationLayerStyleRes = ThemeSwitcher.retrieveNavigationViewStyle(context,
-      R.attr.navigationViewLocationLayerStyle);
-    LocationComponentOptions locationComponentOptions =
-      LocationComponentOptions.createFromAttributes(context, locationLayerStyleRes);
-    locationComponent.activateLocationComponent(context, map.getStyle(), locationComponentOptions);
+    int locationLayerStyleRes = findLayerStyleRes(context);
+    LocationComponentOptions options = LocationComponentOptions.createFromAttributes(context, locationLayerStyleRes);
+    locationComponent.activateLocationComponent(context, map.getStyle(), options);
     locationComponent.setLocationComponentEnabled(true);
     locationComponent.setRenderMode(RenderMode.GPS);
   }
 
+  private int findLayerStyleRes(Context context) {
+    int locationLayerStyleRes = ThemeSwitcher.retrieveNavigationViewStyle(context,
+      R.attr.navigationViewLocationLayerStyle);
+    if (locationLayerStyleRes == INVLAID_LAYER_STYLE) {
```

Comment:
Should we rely on this https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/res/ResourceId.java#38 instead?

Reply:
Previous comment https://github.com/mapbox/mapbox-navigation-android/pull/1696#discussion_r251048084 won't apply if we implement this.